File: spring-java/src/main/java/com/jmpark/springjava/entity/BaseCodeEntity.java
package com.jmpark.springjava.entity;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.ToString;

import javax.persistence.Entity;
import javax.persistence.Id;

@Entity
@RequiredArgsConstructor
@Getter
@ToString
@EqualsAndHashCode(callSuper = false)
public class BaseCodeEntity extends BaseTimeEntity {
    @Id
    String code = "";

    String data1 = null;
    String data2 = null;
    String data3 = null;
    String data4 = null;
}

File: spring-kotlin/src/main/kotlin/com/jmpark/springkotlin/entity/BaseCodeEntity.kt
package com.jmpark.springkotlin.entity

import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class BaseCodeEntity (
    @Id
    val code : String = "",

    var data1 : String? = null,
    var data2 : String? = null,
    var data3 : String? = null,
    var data4 : String? = null
) : BaseTimeEntity()